### PR TITLE
Add changelog entry for changes to isSomeString.

### PR DESCRIPTION
**Pull request by jmdavis**
This is the changelog entry for https://github.com/dlang/phobos/pull/5291.

### CODE REVIEW

File: changelog/std-traits-issomestring.dd [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+`false`. This is particularly true for $(REF `isNarrowString`, std, traits),
+since it's used almost exclusively in range-based code. The same goes for any
+code which uses $(REF `isSomeString`, std, traits) or
+$(REF `isNarrowString`, std, traits) in template constraints or static ifs but
+has other conditions which make enums fail the constraint or static if, since
+then whether $(REF `isSomeString`, std, traits) or
+$(REF `isNarrowString`, std, traits) is `true` or `false` for enums is
+irrelevant.
+
+Code that uses $(REF `isSomeString`, std, traits) or
```
Comment (JackStouffer) [inline changelog/std-traits-issomestring.dd]: This paragraph is redundant.

Comment (jmdavis) [reply changelog/std-traits-issomestring.dd]: Updated.

File: changelog/std-traits-issomestring.dd [hunk 2]
```diff
@@ -0,0 +1,88 @@
+$(REF `isSomeString`, std, traits) and $(REF `isNarrowString`, std, traits) are now `false` for enums
```
Comment (aG0aep6G) [inline changelog/std-traits-issomestring.dd]: Don't use `$(REF ...)` in the summary line. The whole line is turned into a link. REF messes with that.

File: changelog/std-traits-issomestring.dd [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Previously, enums whose base type was a string type were `true` for
+$(REF `isSomeString`, std, traits) and $(REF `isNarrowString`, std, traits).
```
Comment (aG0aep6G) [inline changelog/std-traits-issomestring.dd]: Don't use backticks in `$(REF ...)`. REF already formats as code, and the backticks mess with the URL.

File: changelog/std-traits-issomestring.dd [hunk 4]
```diff
@@ -0,0 +1,88 @@
+isSomeString and isNarrowString are now `false` for enums
+
+Previously, enums whose base type was a string type were `true` for
+$(REF isSomeString, std, traits) and $(REF isNarrowString, std, traits).
+Occasionally, this was useful, but in general, it was a source of bugs, because
+code that works with strings does not necessarily work with enums whose base
+type is a string, making it easy to write code where an enum would pass the
+template constraint and then the template would fail to compile. For instance,
+enums of base type string are `false` for
+$(REF `isInputRange`, std, range, primitives). As such, it really doesn't make
```
Comment (aG0aep6G) [inline changelog/std-traits-issomestring.dd]: You missed one here. https://github.com/dlang/phobos/pull/6108

**Comment by wilzbach** [comment]
Thanks a lot for pointing out the use of REF in the title @aG0aep6G!
@JackStouffer and all: you are aware that you can (and should) give the rendered changelog a quick glance at DAutoTest - it takes ten seconds and can prevent such problems ;-)

**Pull request merged**